bindings/rust/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+#![doc = include_str!("../README.md")]
+#![warn(missing_debug_implementations, missing_docs, rustdoc::all)]
+#![deny(unused_must_use, rust_2018_idioms)]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+
```

Comment:
Can we make this crate `no_std` and switch the `std::collections::HashMap` data structure to use the `no_std` [hashbrown](https://docs.rs/hashbrown/latest/hashbrown/) crate like `kona` uses (`hashbrown = "0.14.3"`).

```suggestion
#![no_std]

```